GitHub Actions: Add Release workflow (#691)
NVIDIA/k8s-device-plugin: RELEASE.md

- The Device Plugin helm chart

Publishing the container is automated through gitlab-ci and only requires one to tag the commit and push it to gitlab.

# Release Process Checklist
- [ ] Run the `./hack/prepare-release.sh` script to update the version in all the needed files
- [ ] Run the `./hack/package-helm-charts.sh` script to generate the helm charts
- [ ] Run the `./hack/generate-changelog.sh` script to generate the a draft changelog
- [ ] Update the CHANGELOG.md file with the generated changelog
- [ ] Commit, Tag and Push to Gitlab
- [ ] Wait for the `Release` GitHub Action to complete
- [ ] Publish the [draft release](https://github.com/NVIDIA/k8s-device-plugin/releases) created by the GitHub Action
- [ ] Wait for the `Publish Helm Chart` GitHub Action to complete

## Troubleshooting
- If the `Release` GitHub Action fails:
    - check the logs for the error first.
    - Manually run the `Release` GitHub Action locally with the same inputs.

>     - If the action continues to fail, manually run `./hack/package-helm-charts.sh` and updload the generated charts to the release assets.

Reviewer: ```suggestion
    - If the action fails, manually run `./hack/package-helm-charts.sh` and upload the generated charts to the release assets.
```

- If the `Publish Helm Chart` GitHub Action fails:

>     - check the logs for the error first.

Reviewer: ```suggestion
    - Check the logs for the error.
```

- [ ] Run the `./hack/generate-changelog.sh` script to generate the a draft changelog
- [ ] Update the CHANGELOG.md file with the generated changelog

> - [ ] Commit, Tag and Push to Gitlab

Reviewer 2: It seems as if some fo the steps were lost somewhere. We no longer mention the steps to publish the image. Could we readd those?

Author: I don't follow, current https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/main/RELEASE.md RELEASE doesn't hold steps for image release. ALl we have is line 7
```md
Publishing the container is automated through gitlab-ci and only requires one to tag the commit and push it to gitlab.
```

Reviewer 2: Sorry, I think I'm misremembering the steps we outlined in github issues / JIRA and the steps in this file. (see for example https://github.com/NVIDIA/cloud-native-team/issues/59).

The `Commit, Tag and Push` to Gitlab is the image trigger though, so we may want to reword this.

Not a blocker for this PR though.

- [ ] Wait for the `Release` GitHub Action to complete

> - [ ] Create a [new release](https://github.com/NVIDIA/k8s-device-plugin/releases) from the draft release created by the GitHub Action

Reviewer 2: Nit:
```suggestion
- [ ] Publish the [draft release](https://github.com/NVIDIA/k8s-device-plugin/releases) created by the GitHub Action
```

- [ ] Wait for the `Publish Helm Chart` GitHub Action to complete

## Troubleshooting
- If the `Release` GitHub Action fails:

>     - check the logs for the error first.

Reviewer 2: Is rerunning the job an option?

>     - Manually run `./hack/package-helm-charts.sh` and updload the generated charts to the release assets

Reviewer 2: Should we list the assets?

Author: is just 2 files with `tgz` sufix. I am not against it. up to you

- [ ] Run the `./hack/generate-changelog.sh` script to generate the a draft changelog
- [ ] Update the CHANGELOG.md file with the generated changelog
- [ ] Commit, Tag and Push to Gitlab
- [ ] Wait for the `Release` GitHub Action to complete

> - [ ] Create a [new release](https://github.com/NVIDIA/k8s-device-plugin/releases)from the draft release created by the GitHub Action

Reviewer 2: Do we want to document how to do this if things go wrong?

Author: done! good idea
